### DISCUSSION TITLE
tw-classed vs css-in-js

### DISCUSSION THREAD

**Discussion by kenbanks-peng** [sannajammeh/tw-classed#118] [Q&A] [answered]
Appreciating both tailwind and css-in-js, I very much like tw-classed. Thank you. 

From various articles (for example,  [here](https://github.com/reactwg/react-18/discussions/110) and [here](https://dev.to/srmagura/why-were-breaking-up-wiht-css-in-js-4g9b), the future of css-in-js for react may not shine as bright as it once did.

Does tw-classes apply the same run-time rendering for styles as css-in-js does or does tw-classed produce static css via tailwind?

Thank you.

**Comment by sannajammeh** [comment, marked as answer]
TW Classed is nothing more than a class name toggler. All CSS generation is handled via Tailwind as usual. This opens up a lot of doors in terms of both performance and usage, for example, I use `tw-classed` in React Native to toggle Nativewind classes.

TW Classed is also universal, it can be used to toggle CSS modules, regular class names, tailwind and UnoCSS. 

See https://tw-classed.vercel.app/docs/benchmark for comparison to Stitches (the fastest css-in-js library)

**Reply by kenbanks-peng** [reply]
Most appreciated.

**Reply by Seanmclem** [reply]
Would it work in SSG or SSR? Astro?

**Reply by sannajammeh** [reply]
> Would it work in SSG or SSR? Astro?

It works with all 3, even React server components! It just a more convenient way to add classes to components 😃
